packages/next-intl/plugin.js

```diff
@@ -39,7 +39,7 @@ module.exports = withNextIntl({
     }
   }
 
-  if ('i18n' in nextConfig) {
+  if (nextConfig.i18n != null) {
```

Comment:
The change looks good, but I don't really understand what you mean by "dynamically constructing" the Next.js configuration. Can you elaborate?

Reply:
I verified the change locally, looks good!

Reply:
As explained in the [docs](https://nextjs.org/docs/api-reference/next.config.js/introduction), in your `next.config.js` module there are two supported configuration methods. Either the module exports as `default` `NextConfig` object, or a function accepting a `phase: string` (possible literal strings are found in `next/dist/shared/lib/constants`) and a `config: { defaultConfig: NextConfig }`.

The function export is useful for dynamically populating the Next.js configuration depending on the current `phase`. At my workplace we're using this functionality to control the headers Next.js serves, which is useful for selectively disabling the [content security policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) headers in the development environment (as it's a pain to get them working with `http://localhost` and we'd rather not forget to enable them for our production builds). We also perform validation of our environmental variables (using [zod](https://github.com/colinhacks/zod)) and add them to the `ProcessEnv` NodeJS typing namespace within this function, that way Next.js is prevented from starting up until the issue is resolved (as opposed to throwing these errors during runtime, which is easier to overlook and harder to debug).

The issue that part of the PR resolves is that Next.js provides a `defaultConfig` object with the `i18n` property initialised as `null`, since the `next-intl` plugin module was issuing warnings based on the presence of the field (rather than its value) it was incorrectly throwing out warnings.

The other issue was found as we enable type-checking in the `next.config.js` module. Might be worth adding a test to `next-intl` which also does this, otherwise it's unlikely you'll catch any issues with parts of the interface consumed by non-Typescript code. The process how to do this well documented within the Typescript documentation section on [type checking javascript files](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html).

Reply:
Anyways sorry for the infodump, couldn't help myself 😊

Reply:
Hey, thanks a lot for the extensive answer! That's what I thought is happening with the `phase`, but I've never used it personally! 🙂 

You can upgrade to `next-intl@2.14.0-beta.2` to get rid of the issue!

Reply:
Thank you very much, I can yeet my `delete config.i18n` line now!